resources/00-taisei.pkgdir/shader/stage1_water.frag.glsl

```diff
@@ -12,8 +12,13 @@ float smoothNoise(vec2 p) {
 	p -= f;
 	f *= f * (3 - f - f);
 
+	// Intel's sin() function breaks down with values over 1e4 (yes, really)
+	// get rid of some bits here w/o sacrificing the nice effect
+	vec4 a = vec4(0, 1, 27, 28) + p.x + p.y * 27;
+	vec4 b = a * mod(1e5, tau);
+
 	// WARNING: Some versions of the Windows AMD driver choke on temp_mat = mat2(temp_vec)
-	vec4 temp_vec = fract(sin(vec4(0, 1, 27, 28) + p.x + p.y * 27) * 1e5);
+	vec4 temp_vec = fract(sin(b));
```

Comment:
Wait, now I'm actually confused by this. In the original code:
```glsl
fract(sin(vec4(0, 1, 27, 28) + p.x + p.y * 27) * 1e5)
```
The 1e5 factor is actually outside of the sin's argument... i.e.
```glsl
vec4 sin_arg = vec4(0, 1, 27, 28) + p.x + p.y * 27;
vec4 temp_vec = fract(sin(sin_arg) * 1e5);
```

But in the rewritten form it goes inside the sin. And ` mod(1e5, tau)` is actually a constant (`~3.10583...`). What gives?